tests/test_shell_launcher.py

```diff
@@ -0,0 +1,323 @@
+        open(out_file, "w", encoding="utf-8") as out,
+        open(err_file, "w", encoding="utf-8") as err,
+    ):
+        args = (helpers.expand_exe_path("srun"), "--flag_dne")
+        cmd = ShellLauncherCommand({}, run_dir, out, err, args)
+        id = shell_launcher.start(cmd)
+        proc = shell_launcher._launched[id]
+        proc.wait()
+        assert proc.returncode != 0
+        with open(out_file, "r", encoding="utf-8") as out:
+            assert out.read() == ""
+        with open(err_file, "r", encoding="utf-8") as err:
+            content = err.read()
+            assert "unrecognized option" in content
+
+
+def test_popen_issues_unique_ids(shell_launcher, test_dir):
+    """Validate that all ids are unique within ShellLauncher._launched"""
+    run_dir, out_file, err_file = generate_directory(test_dir)
+    with (
+        open(out_file, "w", encoding="utf-8") as out,
+        open(err_file, "w", encoding="utf-8") as err,
+    ):
+        cmd = ShellLauncherCommand(
+            {}, run_dir, out, err, EchoHelloWorldEntity().as_program_arguments()
+        )
+        for _ in range(5):
+            _ = shell_launcher.start(cmd)
+        assert len(shell_launcher._launched) == 5
+        assert all(proc.wait() == 0 for proc in shell_launcher._launched.values())
+
+
+def test_retrieve_status_dne(shell_launcher):
+    """Test tht ShellLauncher returns the status of completed Jobs"""
+    # Init ShellLauncher
+    with pytest.raises(LauncherJobNotFound):
+        _ = shell_launcher.get_status("dne")
+
+
+def test_shell_launcher_returns_complete_status(shell_launcher, test_dir):
+    """Test tht ShellLauncher returns the status of completed Jobs"""
+    run_dir, out_file, err_file = generate_directory(test_dir)
+    with (
+        open(out_file, "w", encoding="utf-8") as out,
+        open(err_file, "w", encoding="utf-8") as err,
+    ):
+        cmd = ShellLauncherCommand(
+            {}, run_dir, out, err, EchoHelloWorldEntity().as_program_arguments()
+        )
+        for _ in range(5):
+            id = shell_launcher.start(cmd)
+            proc = shell_launcher._launched[id]
+            proc.wait()
+            code = shell_launcher.get_status(id)[id]
+            assert code == JobStatus.COMPLETED
+
+
+@requires_slurm
+def test_shell_launcher_returns_failed_status(shell_launcher, test_dir):
+    """Test tht ShellLauncher returns the status of completed Jobs"""
+    run_dir, out_file, err_file = generate_directory(test_dir)
+    with (
+        open(out_file, "w", encoding="utf-8") as out,
+        open(err_file, "w", encoding="utf-8") as err,
+    ):
+        args = (helpers.expand_exe_path("srun"), "--flag_dne")
+        cmd = ShellLauncherCommand({}, run_dir, out, err, args)
+        for _ in range(5):
+            id = shell_launcher.start(cmd)
+            proc = shell_launcher._launched[id]
+            proc.wait()
+            code = shell_launcher.get_status(id)[id]
+            assert code == JobStatus.FAILED
```

Comment:
Ditto "does this strictly need to be `srun` question here"

Reply:
not at all actually!